### PR TITLE
Overlays finalization

### PR DESCRIPTION
**Pull request by mertwole**

### CODE REVIEW

File: JamGame/Assets/Scripts/Overlay/OverlayManager.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+using UnityEngine;
+
+public interface IOverlayManager
+{
+    public void RevertAllOverlays();
+
+    public void ApplyOverlay<O>(O overlay) where O : class, IOverlay;
+}
+
+public class OverlayManager : MonoBehaviour, IOverlayManager
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Overlay/OverlayManager.cs:12]: Really? Manager?

File: JamGame/Assets/Scripts/Overlay/OverlayManager.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{
+    // TODO: Update in runtime.
+    private List<IOverlayRenderer> overlayRenderers;
+
+    private void Start()
+    {
+        overlayRenderers = GetComponentsInChildren<IOverlayRenderer>().ToList();
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/Overlay/OverlayManager.cs:19]: For sure, it do not will get components in disable objects or if component disables

File: JamGame/Assets/Scripts/UI/Overlays/OverlaySelector/OverlaySelector.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    void Start()
+    {
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
```
Comment (Kiuh) [inline JamGame/Assets/Scripts/UI/Overlays/OverlaySelector/OverlaySelector.cs:17]: wasting code lines

**Review by Kiuh — APPROVED** [review]

File: JamGame/Assets/Scripts/Overlay/OverlayManager.cs (hunk 1 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Overlay/OverlayManager.cs:12]: any suggestions?

File: JamGame/Assets/Scripts/Overlay/OverlayManager.cs (hunk 2 above)
Comment (mertwole) [reply JamGame/Assets/Scripts/Overlay/OverlayManager.cs:19]: When Employees will spawn in location they will notify OverlayManager about this

Comment (mertwole) [reply JamGame/Assets/Scripts/Overlay/OverlayManager.cs:19]: The current approach is for testing purposes mainly

File: JamGame/Assets/Scripts/Overlay/OverlayManager.cs (hunk 1 above)
Comment (Kiuh) [reply JamGame/Assets/Scripts/Overlay/OverlayManager.cs:12]: Maybe switcher or Applier, if no - leave it as it is

**Review by Kiuh — COMMENTED** [review]
And why `NeedProvider.cs`  in Scripts root folder?

**Pull request merged**